nucypher/cli/commands/enrico.py

```diff
@@ -63,25 +63,38 @@ def run(general_config, policy_encrypting_key, dry_run, http_port):
 def encrypt(general_config, policy_encrypting_key, message, file, ipfs):
     """Encrypt a message under a given policy public key."""
 
-    # Setup
     emitter = setup_emitter(general_config=general_config, banner=policy_encrypting_key)
+
+    if ipfs:
+        try:
+            import ipfshttpclient
+        except ImportError:
+            raise ImportError("IPFS HTTP client not installed.  Run pip install ipfshttpclient then try again.")
+
+        # Connect to IPFS before proceeding
+        ipfs_client = ipfshttpclient.connect(ipfs)
+        emitter.message(f"Connected to IPFS Gateway {ipfs}")
+
+    if not policy_encrypting_key:
+        policy_encrypting_key = click.prompt("Enter policy public key", type=click.STRING)
+
     ENRICO = _create_enrico(emitter, policy_encrypting_key)
-    if not (bool(message) ^ bool(file)):
-        emitter.error(f'Pass either --message or --file. Got {message}, {file}.')
+    if message and file:
+        emitter.error(f'Pass either --message or --file, not both.')
         raise click.Abort
 
+    if not message:
+        message = click.prompt('Enter plaintext to encrypt', type=click.STRING)
+
```

Comment:
confused here - what if `file` was passed; doesn't this nullify the original XOR check?

Reply:
Unsure - this is more or less an "application level" decision.  What's a sensible default behavior here?

Reply:
I guess my thought was that the original code ensured that only one of them is set by using the XOR - `if not (bool(message) ^ bool(file)):`. In this change, we ensure that both aren't set, `if message and file` but we don't know which one, `file` or `message` is actually set, or if they are set at all.

Subsequently, you check on L86 whether `message` is not set, which could mean that either `file` is or is not set`. I think to do what you are trying to do here you might have to do something like:

```suggestion
    if not file and not message:
        message = click.prompt('Enter plaintext to encrypt', type=click.STRING)

```

Reply:
> I guess my thought was that the original code ensured that only one of them is set by using the XOR

It doesn't play nicely with the interactivity -  When might this check be effective if we're prompting for manual input?

>  I think to do what you are trying to do here you might have to do something like

I'm liking this simple approach here, with the default behavior focused around encrypting messages (good for first-time users, hobbyist, demoers).  `--file` can be added if needed.

```suggestion
    if not message or file:
        message = click.prompt('Enter plaintext to encrypt', type=click.STRING)

```

Reply:
:heavy_check_mark: